src/main/java/edu/harvard/iq/dataverse/api/Search.java

```diff
@@ -193,24 +193,8 @@ public Response search(
     }
 
     private User getUser() throws WrappedResponse {
-        /**
-         * @todo support searching as non-guest:
-         * https://github.com/IQSS/dataverse/issues/1299
-         *
-         * Note that superusers can't currently use the Search API because they
-         * see permission documents (all Solr documents, really) and we get a
-         * NPE when trying to determine the DvObject type if their query matches
-         * a permission document.
-         *
-         * @todo Check back on https://github.com/IQSS/dataverse/issues/1838 for
-         * when/if the Search API is opened up to not require a key.
-         */
-        AuthenticatedUser authenticatedUser = findAuthenticatedUserOrDie();
-        if (nonPublicSearchAllowed()) {
-            return authenticatedUser;
-        } else {
-            return GuestUser.get();
-        }
+        return settingsSvc.isTrueForKey(SettingsServiceBean.Key.SearchApiRequireToken, false) ? 
+                findAuthenticatedUserOrDie() : findUserOrDie();
```

Comment:
Depends on your style. I like it since it makes more explicit the fact that there's a single value being decided on based on a condition.
I normally try to reserve `if {..} else {..}` (which is a statement rather than an expression) to different paths in an algorithm.

So, literature-like style reasons, no hard data here.